### PR TITLE
Update git-hg

### PR DESCRIPTION
**Pull request by rochmadadhiyunianto**
#85 #82 Fix Shellcheck on git-hg

### CODE REVIEW

File: git-hg/add-branch.sh [hunk 1]
```diff
@@ -12,12 +13,12 @@ bpath=$1
 
 echo "Create $bpath"
 
-mkdir -p $bpath || exit 1
+mkdir -p bpath || exit 1
```
Comment (karianna) [inline git-hg/add-branch.sh]: `$` removed in error?

File: git-hg/add-branch.sh [hunk 2]
```diff
@@ -12,12 +13,12 @@ bpath=$1
...
 
 echo "Clone $bpath (root)"
 git hg clone "http://hg.openjdk.java.net/$bpath" "$bpath/root" || exit 1
 
-for m in $submodules
+for m in "$@submodules"
```
Comment (karianna) [inline git-hg/add-branch.sh]: `m` --> `module`

File: git-hg/update.sh [hunk 3]
```diff
@@ -19,7 +21,7 @@ git hg fetch "http://hg.openjdk.java.net/$bpath"
 git hg pull "http://hg.openjdk.java.net/$bpath"
 popd
 
-for m in $submodules
+for m in "$@submodules"
 do
```
Comment (karianna) [inline git-hg/update.sh]: `m` --> `module`

**Comment by UcupBS** [comment]
Hi,, I just revert my typo using old script, but adding shellscript disable reference. and I use https://www.shellcheck.net/ to check it

**Comment by karianna** [comment]
@UcupBS I think you were actually on the right course previously, it was just the one typo which I think was the issue (removing the `$` from bpath).  I don't think disabling Shellcheck is the right thing to do here...

**Comment by butuh** [comment]
please check new one..

**Comment by karianna** [comment]
@butuh / @UcupBS - I've added some extra comments and @gdams will need to review as well.

File: git-hg/add-branch.sh [hunk 4]
```diff
@@ -2,63 +2,63 @@
...
 
     echo "Merge '$m'"
-    git subtree add --prefix=$m imports/$bpath/$m/master -m "Initial merge of '$m'" || exit 1
+    git subtree merge --prefix="$m" imports/"$bpath"/"$m"/master -m "Merge from '$m'" || exit 1
 done
 
 echo "Push"
 
-git push github master --tags || exit 1
+git push github master --tags
```
Comment (gdams) [inline git-hg/add-branch.sh]: any reason that you dropped the `|| exit 1` here?

Comment (UcupBS) [inline git-hg/add-branch.sh]: I dont have reason for this.. because I check on shellcheck dont get any error

**Comment by karianna** [comment]
@butuh / @UcupBS - there's still a couple of outstanding comments to fix, are you able to update the PR?

File: git-hg/add-branch.sh [hunk 5]
```diff
@@ -2,63 +2,63 @@
...
 
+# shellcheck disable=SC1091
 . import-common.sh
 
 echo "Enter hg"
 
 cd hg || exit 1
 
-bpath=$1
+echo "bpaths=(ls -d -1 */*)"
```
Comment (karianna) [inline git-hg/add-branch.sh]: I'm not sure that `ls -d -1 */*` is what you want to do here?  That lists all files and dirs (`-d` doesn't quite do what you might expect?)

File: git-hg/add-branch.sh [hunk 6]
```diff
@@ -2,63 +2,63 @@
...
+for bpath in "$@bpaths"
+do
 
-echo "Clone $bpath (root)"
-git hg clone "http://hg.openjdk.java.net/$bpath" "$bpath/root" || exit 1
+pushd "$bpath/root"
+echo "Update $bpath -> (root)"
+git hg fetch "http://hg.openjdk.java.net/$bpath"
+git hg pull "http://hg.openjdk.java.net/$bpath"
+popd
```
Comment (karianna) [inline git-hg/add-branch.sh]: Are you missing a `done` here?

File: git-hg/add-branch.sh [hunk 7]
```diff
@@ -2,63 +2,63 @@
...
 
-echo "Create $bpath"
-
-mkdir -p $bpath || exit 1
+for bpath in "$@bpaths"
+do
 
-echo "Clone $bpath (root)"
-git hg clone "http://hg.openjdk.java.net/$bpath" "$bpath/root" || exit 1
+pushd "$bpath/root"
```
Comment (karianna) [inline git-hg/add-branch.sh]: code inside the for loop needs whitespace indentation

File: git-hg/add-branch.sh [hunk 8]
```diff
@@ -2,63 +2,63 @@
...
+echo "Update $bpath -> (root)"
+git hg fetch "http://hg.openjdk.java.net/$bpath"
+git hg pull "http://hg.openjdk.java.net/$bpath"
+popd
 
-for m in $submodules
+for m in "$@submodules"
```
Comment (karianna) [inline git-hg/add-branch.sh]: m --> module

File: git-hg/add-branch.sh [hunk 9]
```diff
@@ -2,63 +2,63 @@
...
 
 echo "Merge (root)"
 
-#git merge imports/$bpath/root/master -m "Initial merge of (root)" --no-ff || exit 1
-git merge imports/$bpath/root/master -m "Initial merge of (root)" || exit 1
+git merge imports/"$bpath"/root/master -m "Merge from (root)" --no-ff || exit 1
 
-for m in $submodules
+for m in "$@submodules"
 do
```
Comment (karianna) [inline git-hg/add-branch.sh]: m --> module

File: git-hg/update.sh [hunk 10]
```diff
@@ -2,15 +2,16 @@
...
 
+# shellcheck disable=SC1091
 . import-common.sh
 
 echo "Enter hg"
 
 cd hg || exit 1
 
-bpaths=`ls -d -1 */*`
+echo "bpaths=(ls -d -1 */*)"
```
Comment (karianna) [inline git-hg/update.sh]: Same comment as previous, not 100% sure this is listing what you want?

File: git-hg/update.sh [hunk 11]
```diff
@@ -39,25 +40,25 @@ git checkout master || exit 1
...
+git fetch imports/"$bpath"/root || exit 1
 
 echo "Merge (root)"
 
-git merge imports/$bpath/root/master -m "Merge from (root)" --no-ff || exit 1
+git merge imports/"$bpath"/root/master -m "Merge from (root)" --no-ff || exit 1
 
-for m in $submodules
+for m in "$@submodules"
 do
```
Comment (karianna) [inline git-hg/update.sh]: m --> module

**Comment by karianna** [comment]
@butuh / @UcupBS - Are you able to rebase and update this PR?

**Comment by neomatrix369** [comment]
@UcupBS Please resolve conflicts and also update PR with responses to review comments - thnx

**Comment by karianna** [comment]
@UcupBS @butuh - Are you able to rebase and push?

**Comment by karianna** [comment]
@ucupbs @butuh - We took your fixes and applied manually - thanks!

**Pull request closed without merging**